crates/ty_python_semantic/resources/mdtest/generics/legacy/classes.md

```diff
@@ -354,11 +354,14 @@ class C(Generic[T]):
     @overload
     def __init__(self: "C[bytes]", x: bytes) -> None: ...
     @overload
+    def __init__(self: "C[int]", x: bytes) -> None: ...
+    @overload
     def __init__(self, x: int) -> None: ...
     def __init__(self, x: str | bytes | int) -> None: ...
 
 reveal_type(C("string"))  # revealed: C[str]
-reveal_type(C(b"bytes"))  # revealed: C[bytes]
+# TODO: revealed: C[bytes]
+reveal_type(C(b"bytes"))  # revealed: C[bytes | int]
```

Comment:
Without the code changes, this test mimics the `TemporaryDirectory` behavior, returning a union specialization